packages/utils/src/abi_decoder.ts

```diff
@@ -89,25 +110,94 @@ export class AbiDecoder {
         }
     }
     /**
-     * Add additional ABI definitions to the AbiDecoder
-     * @param abiArray An array of ABI definitions to add to the AbiDecoder
+     * Decodes transaction data for a known ABI.
+     * @param calldata hex-encoded transaction data.
+     * @param contractName used to disambiguate similar ABI's (optional).
+     * @return Decoded transaction data. Includes: function name and signature, along with the decoded arguments.
      */
-    public addABI(abiArray: AbiDefinition[]): void {
+    public tryDecodeCalldata(calldata: string, contractName?: string): TransactionData {
+        const functionSelector = AbiDecoder._getFunctionSelector(calldata);
+        const candidateFunctionInfos = this._functionInfoBySelector[functionSelector];
+        if (_.isUndefined(candidateFunctionInfos)) {
+            throw new Error(`No functions registered for selector '${functionSelector}'`);
+        }
+        const functionInfo = _.find(candidateFunctionInfos, txDecoder => {
```

Comment:
If we have two candidates. One with undefined contractName and one with defined matching contract name - it makes sense to match against the second one. This code will match against the rightmost one.

Reply:
Not sure I follow. At present, if `contractName` is specified by the caller, then this will return the candidate that was registered to this name. If no `contractName` is specified by the caller then the it will return the first instance.